Framework/PythonInterface/plugins/algorithms/GroupBySampleChangerPosition.py

```diff
@@ -0,0 +1,83 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+# pylint: disable=invalid-name,too-many-instance-attributes,too-many-branches,no-init,deprecated-module
+from mantid.simpleapi import GroupWorkspaces
+from mantid.api import (Algorithm, AlgorithmFactory, WorkspaceGroupProperty)
+from mantid.kernel import (Direction, StringArrayProperty)
+
+SAMP_POSITION = {6.0: "btm",
+                 85.5: "mid",
+                 165.0: "top",
+                 "None": "None"}
+
+
+class GroupBySampleChangerPosition(Algorithm):
+    _input_workspace = None
+    _output_prefix = ''
+    _output_suffix = ''
+
+    def category(self):
+        return 'Workflow\\Inelastic;Inelastic\\Indirect'
+
+    def summary(self):
+        return 'Separates a Group workspace into separate groups based on sample changer position.'
+
+    def PyInit(self):
+        # Input properties
+        self.declareProperty(WorkspaceGroupProperty('InputWorkspace', '',
+                                                    direction=Direction.Input),
+                             doc='Workspace group for the workspaces to be sorted.')
+
+        self.declareProperty('OutputGroupPrefix', '',
+                             doc='String to prefix the output groups.')
+        self.declareProperty('OutputGroupSuffix', '',
+                             doc='String to suffix the output groups.')
+
+        self.declareProperty(StringArrayProperty('OutputWorkspaceList', [], direction=Direction.Output),
+                             doc='List of Workspace group for the resulting workspaces.')
```

Comment:
Not sure about the wording of the doc part of this property. Is this a list of workspaces in the group or a list of the different group names?

Reply:
It is a list of the three workspace groups for the three sample changer positions. not sure what the best way of phrasing that is.

Reply:
Maybe make it groups rather than group in the description?